src/openms/source/FILTERING/DATAREDUCTION/Deisotoper.cpp

```diff
@@ -113,13 +135,17 @@ void Deisotoper::deisotopeAndSingleCharge(MSSpectrum& spec,
           {
             // Possible improvement: include proper averagine model filtering. for now start at the second peak to test hypothesis
             // Note: this is a common approach used in several other search engines
-            if (old_spectrum[p].getIntensity() > old_spectrum[extensions.back()].getIntensity())
+            if (use_averagine && (old_spectrum[p].getIntensity() > old_spectrum[extensions.back()].getIntensity()))
```

Comment:
@timosachsenberg again, the commend says "start at the second peak" but the code doesnt do that... maybe add something like `extensions.size() > 1` here?

Reply:
@netz can you add this or change the comment?

Reply:
I think adding this would actually make sense, to have a legacy, decrasing and none model here

Reply:
Should I add an additional flag then?

Also I think there might be different interpretations of the comment.
The compared intensities are from the peaks `old_spectrum[extensions.back()]`
and `old_spectrum[p]`. In the first iteration of this loop, `old_spectrum[extensions.back()]` is the monoisotopic peak (it is added before this loop starts by default) and `old_spectrum[p]` is the closest peak to the monoisotopic mass + 1 . It tests whether the monoisotopic peak has a higher intensity than the second one, so it starts at the second peak, right?

What you are suggesting is to add a mode where the monoisotopic peak can have a lower intensity than the second one. Skipping this first comparison and starting with comparing the second to the third peak. I would consider that "starting with the third peak", because at this point the second peak is already accepted and the third is the one that is tested.
Did I get that right?